flink-core/src/main/java/org/apache/flink/api/common/state/StateDescriptor.java

```diff
@@ -77,18 +80,22 @@
 
 	/** The serializer for the type. May be eagerly initialized in the constructor,
 	 * or lazily once the type is serialized or an ExecutionConfig is provided. */
+	@Nullable
 	protected TypeSerializer<T> serializer;
 
+	/** The type information describing the value type. Only used to lazily create the serializer
```

Comment:
nit: I think this was also a copying error in the original comment, but this is not necessarily a "value", unless we simply thing of all state as a value, in which case I'm fine with this.